integration_tests/test_remote.py

```diff
@@ -0,0 +1,65 @@
+from git_wrapper.repo import GitRepo
+
+
+def test_prune_tags(repo_root):
+    repo = GitRepo(repo_root)
+
+    repo.remote.add("test", f"file://{repo_root}")
+    remotes = repo.remote.names()
+    assert "test" in remotes
+
+    repo.tag.create("test_tag", "master")
+    assert "test_tag" in repo.tag.names()
+
+    repo.tag.push("test_tag", "test")
+
+    repo.git.push("test", "--delete", "test_tag")
+
+    repo.remote.fetch("test", prune=True, prune_tags=True)
+    assert "test_tag" not in repo.tag.names()
```

Comment:
Unfortunately, this doesn't quite work as is: if you remove L18, the test still pass.

I think you will need 2 repositories for this one, repo_root can be one and a second one that clones from it using the file system approach you have on L7 (the [clone tests](https://github.com/release-depot/git_wrapper/blob/master/integration_tests/test_repo.py#L33) might provide inspiration too, although it's okay to use GitPython directly for this particular test here). Then you can make changes to the main repo_root repo directly (no need to push, just delete at will) and see if cloned_repo.fetch() prunes what it should after that.